src/components/vr_cooperation/include/vr_cooperation/commands/gpb_notification.h

```diff
@@ -30,48 +30,76 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_COMMANDS_UNREGISTER_SERVICE_REQUEST_H_
-#define SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_COMMANDS_UNREGISTER_SERVICE_REQUEST_H_
+#ifndef SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_COMMANDS_GPB_NOTIFICATION_H_
+#define SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_COMMANDS_GPB_NOTIFICATION_H_
 
-#include "vr_cooperation/commands/base_command_request.h"
-#include "vr_cooperation/vr_module.h"
+#include "application_manager/message.h"
+#include "vr_cooperation/commands/command.h"
+#include "vr_cooperation/interface/hmi.pb.h"
 
 namespace vr_cooperation {
+class VRModule;
 
 namespace commands {
 
 /**
- * @brief UnRegisterServiceRequest command class
+ * @brief GPB message command class for notifications
  */
-class UnregisterServiceRequest : public BaseCommandRequest {
+class GPBNotification : public Command {
  public:
   /**
-   * @brief UnregisterServiceRequest class constructor
-   * @param parent pointer to VRModule
+   * @brief GPBNotification class constructor
+   * @param parent VRModule pointer
    * @param message Message from mobile
    **/
-  explicit UnregisterServiceRequest(
-      VRModule* parent,
-      const application_manager::MessagePtr& message);
+  GPBNotification(VRModule* parent,
+                  const application_manager::MessagePtr& message);
 
   /**
-   * @brief UnregisterServiceRequest class destructor
+   * @brief GPBNotification class destructor
    */
-  virtual ~UnregisterServiceRequest();
+  virtual ~GPBNotification();
 
   /**
-   * @brief Execute command
+   * @brief GPBNotification on timeout reaction
    */
-  virtual void Execute();
+  virtual void OnTimeout();
 
   /**
-   * @brief This method will be called whenever new event received
+   * @brief run notification
    */
-  virtual void OnEvent();
+  virtual void Run();
+
+ protected:
+  /**
+   * @brief send notification to HMI
+   * @param message GPB message to send
+   */
+  void SendNotification(vr_hmi_api::ServiceMessage& message);
```

Comment:
In this function, I set rpc_type and set correlation_id for this message.
So it can not be const